common/src/Model/Brush.cpp

```diff
@@ -1289,11 +1296,13 @@ namespace TrenchBroom {
             updateFacesFromGeometry(worldBounds, *m_geometry);
 
             if (addFacesToGeometry.brushEmpty()) {
-                throw GeometryException("Brush is empty");
+                return kdl::expected<void, GeometryException>(GeometryException("Brush is empty"));
             } else  if (!addFacesToGeometry.brushValid()) {
-                throw GeometryException("Brush is invalid");
+                return kdl::expected<void, GeometryException>(GeometryException("Brush is invalid"));
```

Comment:
I think the constructor should be deleted and you have to either use a `::success` or `::failure` static method, so it's absolutely clear which type is being constructed. This is how they work in Rust, Swift, etc.

I can already that you need to read the template arguments carefully to figure out which of these constructing a failure and success.

Reply:
See above.